electrum/tests/test_transaction.py

```diff
@@ -917,3 +917,76 @@ def test_spending_op_cltv_p2wsh(self):
+    #Output of Transaction
+    txout1 = PartialTxOutput(scriptpubkey=bfh('76a914389ffce9cd9ae88dcc0631e88a821ffdbe9bfe2688ac'), value=900000000)
+    txout2 = PartialTxOutput(scriptpubkey=bfh('76a9147480a33f950689af511e6e84c138dbbd3c3ee41588ac'), value=87000000)
+
+    def test_check_sighash_types_sighash_all(self):
+        self.txin.sighash=Sighash.SIGHASH_ALL
+        self.txin.pubkeys = [bfh('0307b8ae49ac90a048e9b53357a2354b3334e9c8bee813ecb98e99a7e07e8c3ba3')]
+        privkey = bfh('730fff80e1413068a05b57d6a58261f07551163369787f349438ea38ca80fac6')
+        tx = PartialTransaction.from_io(inputs=[self.txin], outputs=[self.txout1,self.txout2], locktime=self.locktime, version=1, BIP69_sort=False)
+        sig = tx.sign_txin(0,privkey)
+        self.assertEqual('304402206ac44d672dac41f9b00e28f4df20c52eeb087207e8d758d76d92c6fab3b73e2b0220367750dbbe19290069cba53d096f44530e4f98acaa594810388cf7409a1870ce01',
+                        sig)
+
+    def test_check_sighash_types_sighash_none(self):
+        self.txin.sighash=Sighash.SIGHASH_NONE
+        self.txin.pubkeys = [bfh('03b28f0c28bfab54554ae8c658ac5c3e0ce6e79ad336331f78c428dd43eea8449b')]
+        privkey = bfh('11fa3d25a17cbc22b29c44a484ba552b5a53149d106d3d853e22fdd05a2d8bb3')
+        tx = PartialTransaction.from_io(inputs=[self.txin], outputs=[self.txout1,self.txout2], locktime=self.locktime, version=1, BIP69_sort=False)
+        sig = tx.sign_txin(0,privkey)
+        self.assertEqual('3044022068c7946a43232757cbdf9176f009a928e1cd9a1a8c212f15c1e11ac9f2925d9002205b75f937ff2f9f3c1246e547e54f62e027f64eefa2695578cc6432cdabce271502',
+                        sig)
+
+    def test_check_sighash_types_sighash_single(self):
+        self.txin.sighash=Sighash.SIGHASH_SINGLE
+        self.txin.pubkeys = [bfh('034b8113d703413d57761b8b9781957b8c0ac1dfe69f492580ca4195f50376ba4a')]
+        privkey = bfh('77bf4141a87d55bdd7f3cd0bdccf6e9e642935fec45f2f30047be7b799120661')
+        tx = PartialTransaction.from_io(inputs=[self.txin], outputs=[self.txout1,self.txout2], locktime=self.locktime, version=1, BIP69_sort=False)
+        sig = tx.sign_txin(0,privkey)
+        self.assertEqual('3044022059ebf56d98010a932cf8ecfec54c48e6139ed6adb0728c09cbe1e4fa0915302e022007cd986c8fa870ff5d2b3a89139c9fe7e499259875357e20fcbb15571c76795403',
+                        sig)
+
+    def test_check_sighash_types_sighash_all_anyonecanpay(self):
+        self.txin.sighash=Sighash.SIGHASH_ALL|Sighash.SIGHASH_ANYONECANPAY
+        self.txin.pubkeys = [bfh('033400f6afecb833092a9a21cfdf1ed1376e58c5d1f47de74683123987e967a8f4')]
+        privkey = bfh('14af36970f5025ea3e8b5542c0f8ebe7763e674838d08808896b63c3351ffe49')
+        tx = PartialTransaction.from_io(inputs=[self.txin], outputs=[self.txout1,self.txout2], locktime=self.locktime, version=1, BIP69_sort=False)
+        sig = tx.sign_txin(0,privkey)
+        #Note that this test is modified to the new signature rules which were introduced after bip143.
```

Comment:
This is a bit unclear...
I suggest:
```suggestion
        # Note that the expected sig differs from the one in BIP-0143, due to us grinding the R value. see https://github.com/bitcoin/bitcoin/pull/13666
```
(remember to modify other identical comment accordingly)